assignment-2/src/view/SmartQuestionsList.js

```diff
@@ -0,0 +1,35 @@
+import React, {Component} from "react";
+import question from "../model/Questions";
+import QuestionsList from "./QuestionsList";
+import questionsListPresenter from "../presenter/questionsListPresenter";
+import user from "../model/Users";
+
+const mapModelStateToComponentState = modelState => ({
+    questions: modelState.questions,
+});
+
+export default class SmartQuestionsList extends Component{
+    constructor(){
+        super();
+        this.state = mapModelStateToComponentState(question.state);
+        this.listener = modelState => this.setState(mapModelStateToComponentState(modelState));
+        question.addListener("change",this.listener);
+        user.addListener("change",this.listener);
+    }
+
+    render(){
+        return(
+            <QuestionsList 
+            onAskQuestion={questionsListPresenter.onAskQuestion}
+            onChange={questionsListPresenter.onChange}
+            onSearch={questionsListPresenter.onSearch}
+            onLogout={questionsListPresenter.onLogout}
+            questions={this.state.questions} 
+            username={user.state.currentUser.username}/>
```

Comment:
This is a little hacky (and also bugged I would say). It would be better if you would either:

 1. Have two parameters for your `mapModelStateToComponentState`: one for the user state and the other for the question state. This means that you would need two listeners: one for listening to the user model (and passing the received argument + the state pulled directly from the questionModel) and one for the question model.
 2. two parameters for your `mapModelStateToComponentState`, but a single listener. This would mean that you just ignore the parameter received from the listener and always pass to `mapModelStateToComponentState` the latest values from the models.